cli/wallet/nep5.go

```diff
@@ -159,43 +169,47 @@ func getNEP5Balance(ctx *cli.Context) error {
 		return err
 	}
 
-	var token *wallet.Token
 	name := ctx.String("token")
-	if name != "" {
-		token, err = getMatchingToken(ctx, wall, name)
+
+	for k, acc := range accounts {
+		addrHash, err := address.StringToUint160(acc.Address)
 		if err != nil {
-			token, err = getMatchingTokenRPC(ctx, c, addrHash, name)
-			if err != nil {
-				return cli.NewExitError(err, 1)
-			}
+			return cli.NewExitError(fmt.Errorf("invalid account address: %w", err), 1)
+		}
+		balances, err := c.GetNEP5Balances(addrHash)
+		if err != nil {
+			return cli.NewExitError(err, 1)
 		}
-	}
 
-	balances, err := c.GetNEP5Balances(addrHash)
-	if err != nil {
-		return cli.NewExitError(err, 1)
-	}
+		if k != 0 {
+			fmt.Fprintln(ctx.App.Writer)
+		}
+		fmt.Fprintf(ctx.App.Writer, "Account %s\n", acc.Address)
 
-	for i := range balances.Balances {
-		var tokenName, tokenSymbol string
+		for i := range balances.Balances {
+			var tokenName, tokenSymbol string
 
-		asset := balances.Balances[i].Asset
-		if name != "" && !token.Hash.Equals(asset) {
-			continue
-		}
-		token, err := getMatchingToken(ctx, wall, asset.StringLE())
-		if err != nil {
-			token, err = c.NEP5TokenInfo(asset)
-		}
-		if err == nil {
-			tokenName = token.Name
-			tokenSymbol = token.Symbol
-		} else {
-			tokenSymbol = "UNKNOWN"
+			asset := balances.Balances[i].Asset
+			token, err := getMatchingToken(ctx, wall, asset.StringLE())
+			if err != nil {
+				token, err = c.NEP5TokenInfo(asset)
+			}
+			if err == nil {
+				if name != "" && !(token.Name == name || token.Symbol == name || token.Address() == name || token.Hash.StringLE() == name) {
```

Comment:
When can `token.Address()` or `token.Hash` be `== name`?

Reply:
`--token 668e0c1f9d7b70a99dd9e06eadd4c784d641afbc` is absolutely valid. Address is a bit more wild thing to have, but for a deployed payable contract it can be a thing.